app.go

```diff
@@ -804,6 +810,25 @@ func (app *App) Done() <-chan os.Signal {
 	return c
 }
 
+// Shutdown captures a signal and exit code.
+type Shutdown struct {
+	Signal   os.Signal
+	ExitCode int
+}
+
+// DoneWithCode returns a channel of the Shutdown struct to block on after starting
+// the application. This behaves exactly as Done() does with the addition of
+// capturing the application's exit code, if one is set.
+func (app *App) DoneWithCode() <-chan Shutdown {
```

Comment:
Can we find an alternative name for this? DoneWithCode isn't enough because we're returning a whole Shutdown struct that can conceivably contain more information in the future.

I can't think of anything obvious yet. Maybe "Wait" or similar? We can discuss this.